### PR TITLE
Releasing process - use signed tags

### PR DESCRIPTION
**Pull request by Kielek**
Fixes #.

## Changes

Use signed tags while releasing packages.
It propagate good practice from autoinstrumentation repository: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/f5c675ecc69c0408e3dc21915a3208a04c8c472e/docs/releasing.md?plain=1#L37

Effect on GitHub side (left unsigned/right signed): 
image

For significant contributions please make sure you have completed the following items:

* ~~[ ] Appropriate `CHANGELOG.md` updated for non-trivial changes~~
* ~~[ ] Design discussion issue #~~
* ~~[ ] Changes in public API reviewed~~

### CODE REVIEW

**Review by pjanotti — APPROVED** [review]

**Review by reyang — APPROVED** [review]

File: build/RELEASING.md
Comment (cijothomas) [inline build/RELEASING.md:41]: will ask all maintainers to test and confirm before merge.

**Review by pellared — APPROVED** [review]

File: build/RELEASING.md
Comment (alanwest) [reply build/RELEASING.md:41]: This will require maintainers to create and manage their own GPG key. I'd be fine with this as I've been doing it for years, but most people I work with have not adopted this practice. Do other maintainers want to do this?

https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key

Comment (utpilla) [reply build/RELEASING.md:41]: > This will require maintainers to create and manage their own GPG key.

This is fine with me as well.

Comment (Kielek) [reply build/RELEASING.md:41]: @CodeBlanch, @cijothomas, what do you think?

**Pull request closed without merging**
